SwiftCompilerSources/Sources/SIL/Function.swift

```diff
@@ -301,6 +301,18 @@ final public class Function : CustomStringConvertible, HasShortDescription, Hash
           }
         }
         return BridgedEffectInfo(argumentIndex: -1, isDerived: false, isEmpty: true, isValid: false)
+      },
```

Comment:
We don't need this.
For the purpose of deinit barriers, we just need to bridge `Instruction.isDeinitBarrier(_ analysis: CalleeAnalysis)` to make this (and only this) API available in C++.
You can chose where to "register" this API. I would say, it's better done in `CalleeAnalysis`, but I don't have a strong opinion.

Now, for _other_ side effects, we need to add such an API to get effects for an apply (but not for a function). I already have a prototype implementation (https://github.com/eeckstein/swift/tree/alias-analysis), which is similar what you did here.
But this is not related to deinit barriers and you don't need to do it in this PR.

Reply:
Done.